packages/service-provider-node-driver/src/node-driver-service-provider.ts

```diff
@@ -472,40 +472,52 @@ export class NodeDriverServiceProvider
   }
 
   async getConnectionInfo(): Promise<ConnectionInfo> {
-    const [buildInfo = null, atlasVersion = null, fcv = null, atlascliInfo] =
+    const buildInfoPromise = this.runCommandWithCheck(
+      'admin',
+      { buildInfo: 1 },
+      this.baseCmdOptions
+    ).catch(() => {});
+
+    const isLocalAtlasPromise = this.countDocuments('admin', 'atlascli', {
+      managedClusterType: 'atlasCliLocalDevCluster',
+    }).then(
+      (result) => !!result,
+      () => false
+    );
+
+    const atlasVersionPromise = this.runCommandWithCheck(
+      'admin',
+      { atlasVersion: 1 },
+      this.baseCmdOptions
+    ).then(
+      (response) =>
+        typeof response.atlasVersion === 'string'
+          ? response.atlasVersion
+          : undefined,
+      () => undefined
+    );
+
+    const resolvedHostname = this._getHostnameForConnection(
+      this._lastSeenTopology
+    );
```

Comment:
Why can't we use `Promise.all()` on the promises necessary for `getConnectExtraInfo()`, then pass the resolved values there?

If you do want to store promises separately, you'll want to add `.catch()` handlers so that they don't result in unhandled rejections that bring down the process in case they reject

Reply:
I made a backup branch which I pushed 🙈 Now, I pushed to the right one.